File: Examples/CSharp/PerformingandManagingOCR/GettingNotification.cs
using System.IO;
using Aspose.OCR;
using System;

namespace Aspose.OCR.Examples.CSharp.PerformingandManagingOCR
{
    public class GettingNotification
    {
        public static void Run()
        {
            // ExStart:GettingNotification   
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_OMR();
            // Initialize an instance of OcrEngine
            OcrEngine ocrEngine = new OcrEngine();

            // Set the Image property by loading the image from file path location or an instance of MemoryStream 
            ocrEngine.Image = ImageStream.FromFile(dataDir+ "Sampleocr.bmp");

            // Get an instance of INotifier
            INotifier processorWord = NotifierFactory.WordNotifier();

            //Write a delegate to handle the Elapsed event
            processorWord.Elapsed += delegate
            {
                // Display the recognized text on screen
                Console.WriteLine(processorWord.Text);
            };

            // Add the word processor to the OcrEngine
            ocrEngine.AddNotifier(processorWord);

            // Process the image
            ocrEngine.Process();
            // ExEnd:GettingNotification               
        }
    }
}

File: Examples/VisualBasic/PerformingandManagingOCR/GettingNotification.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.OCR
Imports System

Namespace Aspose.OCR.Examples.VisualBasic.PerformingandManagingOCR
    Public Class GettingNotification
        Public Shared Sub Run()
            ' ExStart:GettingNotification
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_OCR()
            'Initialize an instance of OcrEngine
            Dim ocrEngine As New OcrEngine()

            'Set the Image property by loading the image from file path location or an instance of MemoryStream 
            ocrEngine.Image = ImageStream.FromFile(dataDir & "Sampleocr.bmp")

            Dim processorWord As INotifier = NotifierFactory.WordNotifier()
            AddHandler processorWord.Elapsed, AddressOf [Delegate]

            ' Add the word processor to the OcrEngine
            ocrEngine.AddNotifier(processorWord)

            'Process the image
            ocrEngine.Process()
            ' ExEnd:GettingNotification

        End Sub

        Private Shared Sub [Delegate](sender As Object, recognizedText As IRecognizedText)
            Console.WriteLine("A word was recognized: " & recognizedText.ToString())
        End Sub

    End Class
End Namespace